Src/Base/AMReX_GpuDevice.H

```diff
@@ -123,7 +123,15 @@ public:
     static std::size_t freeMemAvailable ();
 
 #ifdef AMREX_USE_GPU
+
+#   if defined(AMREX_USE_HIP) && (HIP_VERSION_MAJOR >= 4)
```

Comment:
Note: You could also write `#if defined(AMREX_USE_HIP) && defined(__AMDGCN_WAVEFRONT_SIZE)`
But I decided to use exactly what you implemented in this PR because this will guarantee that we use the new macro together with the latest ROCM versions and not silently switch to the old implementation.

Reply:
Thanks for the note!